### PR TITLE
Deploy NVHPC 22.1

### PR DESCRIPTION
**Pull request by olupton**
ABM-related workaround for 21.11 is needed for 22.1 too, but is no longer needed for `develop` versions.

### CODE REVIEW

**Review by matz-e — DISMISSED** [review]

File: var/spack/repos/builtin/packages/nvhpc/package.py [hunk 1]
```diff
@@ -22,6 +22,10 @@
 #  - package key must be in the form '{os}-{arch}' where 'os' is in the
 #    format returned by platform.system() and 'arch' by platform.machine()
 _versions = {
+    '22.1': {
+        'Linux-aarch64': ('05cfa8c520a34eab01272a261b157d421a9ff7129fca7d859b944ce6a16d2255', 'https://developer.download.nvidia.com/hpc-sdk/22.1/nvhpc_2022_221_Linux_aarch64_cuda_multi.tar.gz'),
+        'Linux-ppc64le': ('9fa9b64fba2c9b287b5800693417d8065c695d18cab0526bad41d9aecc8be2b3', 'https://developer.download.nvidia.com/hpc-sdk/22.1/nvhpc_2022_221_Linux_ppc64le_cuda_multi.tar.gz'),
+        'Linux-x86_64': ('7e4366509ed9031ff271e73327dd3121909902a81ac436307801a5373efaff5e', 'https://developer.download.nvidia.com/hpc-sdk/22.1/nvhpc_2022_221_Linux_x86_64_cuda_multi.tar.gz')},
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/nvhpc/package.py:28]: OK due to global metaprogramming…

**Review by matz-e — DISMISSED** [review]

**Comment by olupton** [comment]
This is going to have NEURON build failures in its current draft because https://github.com/BlueBrain/spack/pull/1407 introduced a workaround that was pegged to NVHPC 21.11.

My preferred solution would be to merge https://github.com/neuronsimulator/nrn/pull/1587, so `develop` doesn't need a workaround, and then modify the recipe to insert a workaround for `neuron@:8.0.0%nvhpc@21.11:`.

**Comment by olupton** [comment]
I launched https://bbpgitlab.epfl.ch/hpc/coreneuron/-/pipelines/32849 manually using this PR's Spack branch and deployment directory.

**Comment by olupton** [comment]
Everything passed there, so I think this is good to go.

File: var/spack/repos/builtin/packages/nvhpc/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/nvhpc/package.py:28]: Are these also going into upstream?

With the next merge, `var/spack/repos` is going to be hard-reset to whatever Spack provides.

**Review by matz-e — APPROVED** [review]

File: var/spack/repos/builtin/packages/nvhpc/package.py (hunk 1 above)
Comment (olupton) [reply var/spack/repos/builtin/packages/nvhpc/package.py:28]: This came from upstream: https://github.com/spack/spack/pull/28288

**Pull request merged**